Jenkinsfile

```diff
@@ -54,97 +55,84 @@ pipeline {
                 }
-                */
             }
         }
         // The full build will be run against all regular branches
         stage('Full Build') {
-            when { expression { env.CHANGE_ID == null } }
-            parallel {
-                stage ('Linux') {
-                    steps {
-                        timeout(time: 1, unit: 'HOURS') {
-                            // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
-                            withMaven(maven: 'Maven 3.3.9', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
-                                sh '''
-                                    unset JAVA_TOOL_OPTIONS
-                                    mvn clean install -B -pl !$ITESTS $DISABLE_DOWNLOAD_PROGRESS_OPTS
-                                    mvn install -B -pl $ITESTS -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
-                                '''
-                            }
-                        }
-                    }
+            when { 
+                expression { env.CHANGE_ID == null } 
+            }
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
+                withMaven(maven: 'Maven 3.3.9', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                    sh '''
+                        unset JAVA_TOOL_OPTIONS
+                        mvn clean install -B -pl !$ITESTS $DISABLE_DOWNLOAD_PROGRESS_OPTS
+                        mvn install -B -pl $ITESTS -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+                    '''
                 }
-                //Commenting out Windows builds until pipeline issues can be resolved
-                /*
-                stage ('Windows') {
-                    agent { label 'server-2016-large' }
-                    steps {
-                        retry(3) {
-                            checkout scm
-                        }
-                        timeout(time: 1, unit: 'HOURS') {
-                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}') {
-                                bat 'mvn clean install -B -pl !%ITESTS% %DISABLE_DOWNLOAD_PROGRESS_OPTS%'
-                                bat 'mvn install -B -pl %ITESTS% -nsu %DISABLE_DOWNLOAD_PROGRESS_OPTS%'
-                            }
+            }
+        }
+
+        stage('Codecov.io upload') {
```

Comment:
I like the way this was done. I was thinking of just moving the codcov command up, but this is cleaner

Reply:
Thanks. I was thinking it might be good to group Codecov and SonarCloud as they do similar things, but for the first iteration it was easier to move the Codecov stage rather than merge stages. Maybe something we can review in the future.